Rules.py

```diff
@@ -315,12 +336,13 @@ def global_rules(world):
     # big key gives backdoor access to that from the teleporter in the north west
     set_rule(world.get_location('Misery Mire - Map Chest'), lambda state: state.has_key('Small Key (Misery Mire)', 1) or state.has('Big Key (Misery Mire)'))
     # in addition, you can open the door to the map room before getting access to a color switch, so this is locked behing 2 small keys or the big key...
-    set_rule(world.get_location('Misery Mire - Main Lobby'), lambda state: state.has_key('Small Key (Misery Mire)', 2) or state.has_key('Big Key (Misery Mire)'))
+    set_rule(world.get_location('Misery Mire - Main Lobby'), lambda state: state.has_key('Small Key (Misery Mire)', 1) or state.has_key('Big Key (Misery Mire)'))
     # we can place a small key in the West wing iff it also contains/blocks the Big Key, as we cannot reach and softlock with the basement key door yet
-    set_rule(world.get_entrance('Misery Mire (West)'), lambda state: state.has_key('Small Key (Misery Mire)', 2) if ((item_name(state, 'Misery Mire - Compass Chest') in ['Big Key (Misery Mire)']) or
-                                                                                                                 (item_name(state, 'Misery Mire - Big Key Chest') in ['Big Key (Misery Mire)'])) else state.has_key('Small Key (Misery Mire)', 3))
+    #set_rule(world.get_entrance('Misery Mire (West)'), lambda state: state.has_key('Small Key (Misery Mire)', 2) if ((item_name(state, 'Misery Mire - Compass Chest') in ['Big Key (Misery Mire)']) or
+    #                                                                                                             (item_name(state, 'Misery Mire - Big Key Chest') in ['Big Key (Misery Mire)'])) else state.has_key('Small Key (Misery Mire)', 3))
     set_rule(world.get_location('Misery Mire - Compass Chest'), lambda state: state.has_fire_source())
-    set_rule(world.get_location('Misery Mire - Big Key Chest'), lambda state: state.has_fire_source())
+    set_rule(world.get_location('Misery Mire - Big Key Chest'), lambda state: state.has_fire_source() and 
+                                                                    (item_name(state, 'Misery Mire - Big Key Chest') == 'Big Key (Misery Mire)' and state.has('Small Key (Misery Mire', 2)) or state.has('Small Key (Misery Mire', 3))
     set_rule(world.get_entrance('Misery Mire (Vitreous)'), lambda state: state.has('Cane of Somaria'))
```

Comment:
I'll need to think about these Mire changes.

Reply:
I ported over the VT logic for Mire and Ice Palace, so that's where this came from. My thinking was that as single entrance dungeons it would be a bit simpler to only consider the item locations and the key requirements being moved from doors to chests should prevent unwinnable games.